sofa-tracer-plugins/sofa-tracer-flexible-plugin/src/main/java/com/alipay/sofa/tracer/plugin/flexible/FlexibleStatJsonReporter.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.tracer.plugin.flexible;
+
+import com.alipay.common.tracer.core.appender.builder.JsonStringBuilder;
+import com.alipay.common.tracer.core.appender.file.LoadTestAwareAppender;
+import com.alipay.common.tracer.core.appender.self.SelfLog;
+import com.alipay.common.tracer.core.appender.self.Timestamp;
+import com.alipay.common.tracer.core.constants.SofaTracerConstant;
+import com.alipay.common.tracer.core.reporter.stat.AbstractSofaTracerStatisticReporter;
+import com.alipay.common.tracer.core.reporter.stat.model.StatKey;
+import com.alipay.common.tracer.core.reporter.stat.model.StatMapKey;
+import com.alipay.common.tracer.core.span.CommonSpanTags;
+import com.alipay.common.tracer.core.span.SofaTracerSpan;
+import com.alipay.common.tracer.core.utils.StringUtils;
+import com.alipay.common.tracer.core.utils.TracerUtils;
+import io.opentracing.tag.Tags;
+
+import java.util.Map;
+
+/**
+ * FlexibleStatJsonReporter for flexible biz tracer
+ *
+ * @author: guolei.sgl (guolei.sgl@antfin.com) 2019/8/2 11:50 AM
+ * @since:
+ **/
+public class FlexibleStatJsonReporter extends AbstractSofaTracerStatisticReporter {
+
+    /***
+     * print builder
+     */
+    private static JsonStringBuilder statJson = new JsonStringBuilder();
```

Comment:
is JsonStringBuilder thread-safe?

Reply:
> is JsonStringBuilder thread-safe?
According to the component segregation

Reply:
so it's not thread-safe.  maybe not `static`